py-rse/package-py.Rmd

```diff
@@ -482,450 +173,562 @@ without affecting the main installation (Figure \@ref(fig:py-rse-package-py-vir
-[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
-Type "help", "copyright", "credits" or "license" for more information.
->>> import zipfpy.check
->>>
-```
+## How should I document my package for others? {#py-rse-package-py-document}
 
-Great!  Let's make sure it works from scratch.  Let's deactivate and
-completely delete the previous environment and try again:
+Writing good documentation does not have to be time-consuming
+and involve learning new tools.
+Here we will mention the bare necessities for making a great documentation
+and then highlight extensions that are useful for more complex projects.
 
-```text
-(test_zipfpy) $ deactivate
-$ rm -r test_zipfpy
-$ virtualenv test_zipfpy_2
-$ source test_zipfpy_2/bin/activate
-(test_zipfpy_2) $ pip install -r requirements.txt
-Installing collected packages: atomicwrites, attrs, zipp, importlib-metadata, more-itertools, pyparsing, six, packaging, pluggy, py, wcwidth, pytest
-Successfully installed atomicwrites-1.3.0 attrs-19.1.0 importlib-metadata-0.18 more-itertools-7.0.0 packaging-19.0 pluggy-0.12.0 py-1.8.0 pyparsing-2.4.0 pytest-4.6.3 six-1.12.0 wcwidth-0.1.7 zipp-0.5.1
-(test_zipfpy_2) $ pip install .
-Processing ...
-Building wheels for collected packages: zipfpy
-  Building wheel for zipfpy (setup.py) ... done
-  Stored in directory: /private/var/folders/44/k2tfp8d12h7ggxc0xz8phfmr0000gn/T/pip-ephem-wheel-cache-ab2atma5/wheels/4b/65/6c/557b5122e27983111d3410a06ef63b7b0f3b848df218bb28ed
-Successfully built zipfpy
-Installing collected packages: zipfpy
-Successfully installed zipfpy-0.1
-(test_zipfpy_2) $ python -m pytest
-=============================================================== test session starts ================================================================
-platform darwin -- Python 3.7.2, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
-rootdir: /Users/pterry/Desktop/zipfpy
-collected 3 items
-
-test_zipf.py ...                                                                                                                             [100%]
-
-============================================================= 3 passed in 0.02 seconds =============================================================
-(test_zipfpy_2)
-```
+Documentations starts with writing clear explanatory docstrings for each function and module,
+which we have already done.
+This is one of the most important steps,
+since it means that others can understand what each function is meant to do,
+without going through the source code.
 
-Success!
+What is missing is an introduction to what our package is meant to do,
+instructions on how to install it,
+and some examples of how it can be used.
+We can include these elements in the README.md.
+An example of a good README.md file for our project would be
 
-## How can I distribute command-line scripts in my Python package? {#py-rse-package-py-distrib-scripts}
 
-Our package now works and can be installed in Python environments.
+``````markdown
```

Comment:
Six dashes?

Reply:
It's because there is a nested code block below. But yes, I guess 4 would suffice =)